cilium/cmd/encrypt_status.go

```diff
@@ -0,0 +1,167 @@
+
+	// EncryptionModeIPsec captures enum value "IPsec"
+	EncryptionModeIPsec string = "IPsec"
+
+	// EncryptionModeWireguard captures enum value "Wireguard"
+	EncryptionModeWireguard string = "Wireguard"
+)
+
+type void struct{}
+
+var (
+	voidType    void
+	countErrors int
+)
+
+var encryptStatusCmd = &cobra.Command{
+	Use:   "status",
+	Short: "Display current state of encryption configurations",
+	Run: func(cmd *cobra.Command, args []string) {
+		common.RequireRootPrivilege("cilium encrypt status")
+		getEncryptionMode()
+	},
+}
+
+func init() {
+	encryptCmd.AddCommand(encryptStatusCmd)
+	command.AddJSONOutput(encryptStatusCmd)
+}
+
+func getXfrmStats() (int, map[string]int) {
+	fs, err := procfs.NewDefaultFS()
+	if err != nil {
+		Fatalf("Cannot get a new proc FS: %s", err)
+	}
+	stats, err := fs.NewXfrmStat()
+	if err != nil {
+		Fatalf("Cannot get XFRM states from proc filesystem: %s", err)
+	}
+	v := reflect.ValueOf(stats)
+	errorMap := make(map[string]int)
+	for i := 0; i < v.NumField(); i++ {
+		name := v.Type().Field(i).Name
+		value := v.Field(i).Interface().(int)
+		if value != 0 {
+			countErrors += value
+			errorMap[name] = value
+		}
+	}
+	return countErrors, errorMap
+}
+
+func countUniqueIpsecKeys() int {
+	// trying to mimic set type data structure
+	// using void data type as struct{} because it does not use any memory
+	keys := make(map[string]void)
+	xfrmStates, _ := netlink.XfrmStateList(0)
+	for _, v := range xfrmStates {
+		keys[string(v.Aead.Key)] = voidType
+	}
+	return len(keys)
+}
+
+func maxSequenceNumber() string {
+	var maxSeqNum string
+	out, err := exec.Command("ip", "xfrm", "state").Output()
+	if err != nil {
+		Fatalf("Cannot get xfrm states: %s", err)
+	}
+	commandOutput := string(out)
+	lines := strings.Split(commandOutput, "\n")
+	regex := regexp.MustCompile("oseq[[:blank:]](0[xX][[:xdigit:]]+)?")
+	for _, line := range lines {
+		matched := regex.FindStringSubmatchIndex(line)
+		if matched != nil {
+			maxSeqNum = line[matched[2]:matched[3]]
+		}
+	}
+	return maxSeqNum
+}
+
+func getEncryptionMode() {
+	params := daemon.NewGetHealthzParamsWithTimeout(timeout)
+	params.SetBrief(&brief)
+	if resp, err := client.Daemon.GetHealthz(params); err != nil {
+		Fatalf("Cannot get daemon encryption status:%s", err)
+	} else {
+		encryptionStatusResponse := resp.Payload.Encryption
+		switch encryptionStatusResponse.Mode {
+		case EncryptionModeIPsec:
+			dumpIPsecStatus()
+		case EncryptionModeWireguard:
+			dumpWireGuardStatus(encryptionStatusResponse)
+		default:
+			fmt.Printf("Encryption:%-26s\n", EncryptionModeDisabled)
+		}
+	}
+}
+func dumpIPsecStatus() {
+	fmt.Printf("Encryption:%-26s\n", EncryptionModeIPsec)
+	// Note : node encryption status will be enable once [pr](https://github.com/cilium/cilium/pull/16519) is merged
+	// fmt.Printf("Node Encryption: <TODO>\n")
+	keys := countUniqueIpsecKeys()
+	oseq := maxSequenceNumber()
+	fmt.Printf("Keys in use:%-26d\n", keys)
```

Comment:
Nit: missing space after the colons.
```suggestion
	fmt.Printf("Keys in use: %-26d\n", keys)
```